pages/listing/[[...productId]].tsx

```diff
@@ -4,21 +4,20 @@ import { SimplePool } from "nostr-tools";
 import parseTags, {
   ProductData,
 } from "../components/utility/product-parser-functions";
-import CheckoutPage from "../components/checkout-page";
+import ListingPage from "../components/listing-page";
 import { getLocalStorageData } from "../components/utility/nostr-helper-functions";
 
-const Checkout = () => {
+const Listing = () => {
   const router = useRouter();
   const [relays, setRelays] = useState([]);
   const [productData, setProductData] = useState<ProductData | undefined>(
     undefined,
   );
 
-  const { productId } = router.query;
-  const productIdString = productId ? productId[0] : "";
+  const productIdString = router.asPath.split("/").pop() || ""; // Extract productId from the actual URL
```

Comment:
Not sure if this is the correct way to use router as it was intended since there is the .query functionality